egd_sale_blanket_order_custom/models/blanket_orders.py

```diff
@@ -0,0 +1,133 @@
+        self.order_plan_ids.unlink()
+        order_plans = []
+        Decimal = self.env["decimal.precision"]
+        prec = Decimal.precision_get("Product Unit of Measure")
+        percent = float_round(1.0 / num_installment * 100, prec)
+        percent_last = 100 - (percent * (num_installment - 1))
+        # Normal
+        for i in range(num_installment):
+            this_installment = i + 1
+            if num_installment == this_installment:
+                percent = percent_last
+            vals = {
+                "installment": this_installment,
+                "plan_date": installment_date,
+                "order_type": "installment",
+                "percent": percent,
+            }
+            order_plans.append((0, 0, vals))
+            installment_date = self._next_date(
+                installment_date, interval, interval_type
+            )
+        self.write({"order_plan_ids": order_plans})
+        return True
+
+    def remove_order_plan(self):
+        self.ensure_one()
+        self.order_plan_ids.unlink()
+        return True
+
+    @api.model
+    def _next_date(self, installment_date, interval, interval_type):
+        installment_date = fields.Date.from_string(installment_date)
+        if interval_type == "month":
+            next_date = installment_date + relativedelta(months=+interval)
+        elif interval_type == "year":
+            next_date = installment_date + relativedelta(years=+interval)
+        else:
+            next_date = installment_date + relativedelta(days=+interval)
+        next_date = fields.Date.to_string(next_date)
+        return next_date
+
+    def _create_sale_order(self):
+        order_plan_id = self._context.get("order_plan_id")
+        if order_plan_id:
+            lines = [
+                (
+                    0,
+                    0,
+                    {
+                        "blanket_line_id": line.id,
+                        "product_id": line.product_id.id,
+                        "date_schedule": line.date_schedule,
+                        "remaining_uom_qty": line.remaining_uom_qty,
+                        "price_unit": line.price_unit,
+                        "product_uom": line.product_uom,
+                        "qty": line.remaining_uom_qty,
+                        "partner_id": line.partner_id,
+                    },
+                )
+                for line in self.line_ids
+            ]
+
+            wizard = (
+                self.env["sale.blanket.order.wizard"]
+                .with_context(active_id=self.id, active_model="sale.blanket.order")
+                .create({"blanket_order_id": self.id, "line_ids": lines})
+            )
+
+            result = wizard.create_sale_order()
+            sale_order_id = result.get("domain", [])[0][2][0]
+            moves = self.env["sale.order"].search([("id", "=", sale_order_id)])
+            plan = self.env["sale.order.plan"].browse(order_plan_id)
+            for move in moves:
```

Comment:
```suggestion
            for order in order:
```

e ai corrige todos os move para order